spec.md

```diff
@@ -105,7 +105,73 @@ We define $\operatorname{SPLIT}_C(X)$ recursively, as follows:
 
 # Tree Construction
 
-TODO
+If sequence $X$ and sequence $Y$ are largely the same,
+$\operatorname{SPLIT}_C$ will produce mostly the same chunks,
+choosing the same locations for chunk boundaries
+except in the vicinity of whatever differences there are
+between $X$ and $Y$.
+
+This has obvious benefits for storage and bandwidth,
+as the same chunks can represent both $X$ and $Y$ with few exceptions.
+But while only a small number of chunks may change,
+the _sequence_ of chunks may get totally rewritten,
+as when a difference exists near the beginning of $X$ and $Y$
+and all subsequent chunks have to “shift position” to the left or right.
+Representing the two different sequences may therefore require space
+that is linear in the size of $X$ and $Y$.
+
+We can do better,
+requiring space that is only _logarithmic_ in the size of $X$ and $Y$,
+by organizing the chunks in a tree whose shape,
+like the chunk boundaries themselves,
+is determined by the content of the input.
+The trees representing two slightly different versions of the same input
+will differ only in the subtrees in the vicinity of the differences.
+
+## Definitions
+
+The “level” $L(X)$ of a sequence $X$ is $Q - T$,
+where $Q$ is the largest integer such that
+
+- $Q <= 32$ and
+- $H(P(X)) \mod 2^Q = 0$
```

Comment:
This isn't quite right; `H` takes the sequence of bytes to compute the hash over, so it should be just the last `W` bytes of `P(X)`.

Also, you should probably talk explicitly about the case where the split is triggered by hitting S_max or the end of input (and thus `Q < T`)